src/main/resources/google_checks.xml

```diff
@@ -347,12 +347,17 @@
       <property name="tokens" value="METHOD_DEF, CTOR_DEF, ANNOTATION_FIELD_DEF, COMPACT_CTOR_DEF"/>
     </module>
     <module name="MissingJavadocMethod">
-      <property name="scope" value="public"/>
+      <property name="scope" value="protected"/>
       <property name="allowMissingPropertyJavadoc" value="true"/>
       <property name="allowedAnnotations" value="Override, Test"/>
       <property name="tokens" value="METHOD_DEF, CTOR_DEF, ANNOTATION_FIELD_DEF,
                                    COMPACT_CTOR_DEF"/>
     </module>
+    <module name="SuppressionXpathSingleFilter">
+      <property name="checks" value="MissingJavadocMethod"/>
+      <property name="query" value="//METHOD_DEF[ancestor::CLASS_DEF
+                                  [./MODIFIERS/LITERAL_PROTECTED]]"/>
+    </module>
```

Comment:
I have used the suppression as per your suggestions @romani and now public/protected methods of protected class are not giving warnings, it seems to be working fine

lemme know if this is the correct xpath query or not

Reply:
` <property name="scope" value="protected"/>`
https://checkstyle.org/checks/javadoc/missingjavadocmethod.html#Properties
https://checkstyle.org/property_types.html#Scope

you need "NOT public" somethig like this (not real xpath) `<property name="query" value="//METHOD_DEF[ancestor::CLASS_DEF[not(./MODIFIERS/LITERAL_PUBLIC])]"/>`

Reply:
I have pushed new changes for you to inspect, pls help to create a working xpath query

Reply:
So conceptually you xpath is targeting what we need, not operator works.
```
$ cat Test.java 
/** some. */
public class Test {

  void testing1() {}

  class InnerClass {
    void testing2() {}
  }

}

$ java -jar checkstyle-10.18.1-all.jar -b "//METHOD_DEF[ancestor::CLASS_DEF [not(./MODIFIERS/LITERAL_PUBLIC)]]" Test.java
COMPILATION_UNIT -> COMPILATION_UNIT [2:0]
`--CLASS_DEF -> CLASS_DEF [2:0]
    `--OBJBLOCK -> OBJBLOCK [2:18]
        |--CLASS_DEF -> CLASS_DEF [6:2]
        |   `--OBJBLOCK -> OBJBLOCK [6:19]
        |       |--METHOD_DEF -> METHOD_DEF [7:4]
```

Reply:
```
$ cat Test.java 
/** some. */
public class Test {

  void testing1() {}

  class InnerClass {
    void testing2() {}
  }

  interface InnerClass {
    void testing2() {}
    int f;
  }

}
rivanov@p5510:/var/tmp$ java -jar checkstyle-10.18.1-all.jar \
  -b "//*[self::METHOD_DEF or self::VARIABLE_DEF][ancestor::INTERFACE_DEF[not(./MODIFIERS/LITERAL_PUBLIC)]]" \
  Test.java
COMPILATION_UNIT -> COMPILATION_UNIT [2:0]
`--CLASS_DEF -> CLASS_DEF [2:0]
    `--OBJBLOCK -> OBJBLOCK [2:18]
        |--INTERFACE_DEF -> INTERFACE_DEF [10:2]
        |   `--OBJBLOCK -> OBJBLOCK [10:23]
        |       |--METHOD_DEF -> METHOD_DEF [11:4]
---------
COMPILATION_UNIT -> COMPILATION_UNIT [2:0]
`--CLASS_DEF -> CLASS_DEF [2:0]
    `--OBJBLOCK -> OBJBLOCK [2:18]
        |--INTERFACE_DEF -> INTERFACE_DEF [10:2]
        |   `--OBJBLOCK -> OBJBLOCK [10:23]
        |       |--VARIABLE_DEF -> VARIABLE_DEF [12:4
```

Reply:
what you need is:
```
$ cat Test.java 
/** some. */
public class Test {

  void testing1() {}

  class InnerClass {
    void testing2() {}
  }

  interface InnerClass {
    void testing2() {}
    int f;
  }

}
rivanov@p5510:/var/tmp$ java -jar checkstyle-10.18.1-all.jar \
  -b "//*[self::METHOD_DEF or self::VARIABLE_DEF][ancestor::*[self::INTERFACE_DEF or self::CLASS_DEF][not(./MODIFIERS/LITERAL_PUBLIC)]]" \
  Test.java
COMPILATION_UNIT -> COMPILATION_UNIT [2:0]
`--CLASS_DEF -> CLASS_DEF [2:0]
    `--OBJBLOCK -> OBJBLOCK [2:18]
        |--CLASS_DEF -> CLASS_DEF [6:2]
        |   `--OBJBLOCK -> OBJBLOCK [6:19]
        |       |--METHOD_DEF -> METHOD_DEF [7:4]
---------
COMPILATION_UNIT -> COMPILATION_UNIT [2:0]
`--CLASS_DEF -> CLASS_DEF [2:0]
    `--OBJBLOCK -> OBJBLOCK [2:18]
        |--INTERFACE_DEF -> INTERFACE_DEF [10:2]
        |   `--OBJBLOCK -> OBJBLOCK [10:23]
        |       |--METHOD_DEF -> METHOD_DEF [11:4]
---------
COMPILATION_UNIT -> COMPILATION_UNIT [2:0]
`--CLASS_DEF -> CLASS_DEF [2:0]
    `--OBJBLOCK -> OBJBLOCK [2:18]
        |--INTERFACE_DEF -> INTERFACE_DEF [10:2]
        |   `--OBJBLOCK -> OBJBLOCK [10:23]
        |       |--VARIABLE_DEF -> VARIABLE_DEF [12:4]

```

please extend it for all token type

Reply:
Done, there's a false positives mentioned here: https://github.com/checkstyle/checkstyle/pull/15631#discussion_r1750565524